### PR TITLE
Index cleanup

### PR DESCRIPTION
**Pull request by scottminter**
DO NOT MERGE THIS YET!

### CODE REVIEW

File: node-datasource/xt/foundation/foundation.js [hunk 1]
```diff
@@ -3,377 +3,448 @@ regexp:true, undef:true, strict:true, trailing:true, white:true */
...
+};
 
-  require("./proto");
-  require("./object");
-  require("./io");
-  require("./exception");
+require("./proto")(X);
+require("./object")(X);
+require("./io")(X);
+require("./exception")(X);
```
[the remark this answers is not in the document]
Comment (scottminter) [reply node-datasource/xt/foundation/foundation.js]: It's a shortened way of calling
```
var proto = require('./proto');
proto(X);
```
If the module being required in exports a method then you can call it directly.  I'll leave a comment and cover stuff like this in the Node class.

**Comment by scottminter** [comment]
DO NOT MERGE.  WIP

**Comment by gilmoskowitz** [comment]
There are significant merge conflicts because https://github.com/xtuple/xtuple/pull/3623 was merged first with wider scope.

**Comment by gilmoskowitz** [comment]
Other changes to simplify the repository to a greater extent take precedence.

**Pull request closed without merging**